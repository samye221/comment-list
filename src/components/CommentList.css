.container {
    display: flex;
    flex-direction: column;

}

.message {
    padding: 15px;
}

.privacy {
    font-weight: 700;
}

.messageList {
    margin: 5%;
}

.buttonGroup {
    display: flex;
    flex-direction: row;
    width: 20vw;
    margin: 20px;
}

.icon {
    margin-left: 5% !important;
}