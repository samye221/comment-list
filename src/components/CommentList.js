import React, {useEffect, useState} from 'react';
import {getList, updateList, deleteMessage} from '../services/api';
import {withStyles, makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Button from '@material-ui/core/Button';
import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import './CommentList.css'

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
  },
  inline: {
    display: 'inline',
  },
}));

const LightTooltip = withStyles((theme) => ({
  tooltip: {
    backgroundColor: theme.palette.common.white,
    color: 'rgba(0, 0, 0, 0.87)',
    boxShadow: theme.shadows[1],
    fontSize: 11,
  },
}))(Tooltip);

const MessageList = (props) => {
  const classes = useStyles();
	const [messages, setMessages] = useState([]);
	const [filter, setFilter] = useState('isPublic');
	let isUpdated = props.location.otherProps && props.location.otherProps.isUpdated;

	useEffect(() => {
		const getData = async() => {
			// calls getList or UpdateList conditionnally according to the 'isUpdated' prop set through react-router in the Comment component
			if (isUpdated) {
				const messages = await updateList();
				console.log('MESSAGE_UPDATED', messages);
				if (messages && messages.length) {
					if (filter === 'isPrivate') {
						const filteredMessages = messages.filter(message => message.isPrivate === true);
						return setMessages(filteredMessages);
					}
					if (filter === 'isPublic') {
						const filteredMessages =  messages.filter(message => message.isPrivate === false);
						return setMessages(filteredMessages);				
					}
					else return setMessages(messages)
				}
				return []
			}

			if (!isUpdated) {
				const messages = await getList();
				console.log('MESSAGE', messages)
				if (messages && messages.length) {
					if (filter === 'isPrivate') {
						const filteredMessages = messages.filter(message => message.isPrivate === true);
						return setMessages(filteredMessages);
					}
					if (filter === 'isPublic') {
						const filteredMessages =  messages.filter(message => message.isPrivate === false);
						return setMessages(filteredMessages);
	
					}
					else return setMessages(messages);
				}
				return []
			}
			
		} 
		getData()
	}, [filter, isUpdated]);

	const toggleFilter = (filter) => {
		setFilter(filter);
		isUpdated = false;
	}

	const redirect = () => props.history.push({pathname: '/newComment'})

	// calls the delete function and calls the original static fake-data 
	const handleDelete = async () => {
		deleteMessage();
		const messages = await getList();
		setMessages(messages)
	}
 
	return (
		<div className="container">
			<ButtonGroup>
				<Button color={filter === 'isPublic'? 'primary': 'default'} onClick={() => toggleFilter('isPublic')}>Public</Button>
				<Button color={filter === 'isPrivate'? 'primary': 'default'} onClick={() => toggleFilter('isPrivate')}>Private</Button>
				<Button color={filter === 'all'? 'primary': 'default'}  onClick={() => toggleFilter('all')}>All</Button>
			</ButtonGroup>
			
			<div className="messageList">
				<List className={classes.root}>
					{messages && messages.length && messages.map((message, index) => (
						<div key={`message_${index}`}>
							<ListItem>
								<div className="message">
									<div className="privacy">
										{message.isPrivate ? 'Private' : 'Public'}
									</div>
									<div>
										{message.message}
									</div>
								</div>
								
							</ListItem>
							<Divider variant="inset" component="li" />
						</div>
						
					))}
				</List>
			</div>
			
			<div className="buttonGroup">
				<Button onClick={() => redirect()} variant="contained" color="primary">
					WRITE A MESSAGE
				</Button>
				<LightTooltip title="Delete new message">
  				<IconButton aria-label="delete" className="icon" onClick={()=> handleDelete()}>
    				<DeleteIcon color="secondary" />
  					</IconButton>
				</LightTooltip>
			</div>
		</div>
	)
} 

export default MessageList;
