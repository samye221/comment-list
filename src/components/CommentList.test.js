import React from 'react';
import { createMemoryHistory, createLocation } from 'history';
import CommentList from './CommentList';
import renderer from 'react-test-renderer'

const match = {
    isExact: false,
    url: "/",
    params: { id: "1" }
};
const history = createMemoryHistory();
const location = createLocation(match.url);

it('renders correctly',() => { 
    const tree = renderer.create(<CommentList history={history} location={location}/>).toJSON()
    expect(tree).toMatchSnapshot()
})