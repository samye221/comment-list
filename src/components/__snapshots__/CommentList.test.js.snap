// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="MuiButtonGroup-root"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined MuiButton-outlinedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Public
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Private
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        All
      </span>
    </button>
  </div>
  <div
    className="messageList"
  >
    <ul
      className="MuiList-root makeStyles-root-1 MuiList-padding"
    >
      0
      <div>
        No data
      </div>
    </ul>
  </div>
  <div
    className="buttonGroup"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        WRITE A MESSAGE
      </span>
    </button>
  </div>
</div>
`;
